### PR TITLE
Centralize build parameters

### PR DESCRIPTION
**Pull request by blindpirate**

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test qf

File: build-logic/basics/src/main/kotlin/gradlebuild/basics/BuildParams.kt [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+import gradlebuild.basics.BuildParams.TEST_SPLIT_EXCLUDE_TEST_CLASSES
+import gradlebuild.basics.BuildParams.TEST_SPLIT_INCLUDE_TEST_CLASSES
+import gradlebuild.basics.BuildParams.TEST_SPLIT_ONLY_TEST_GRADLE_VERSION
+import org.gradle.api.JavaVersion
+import org.gradle.api.Project
+import org.gradle.api.provider.Provider
+
+
+object BuildParams {
+    const val BUILD_BRANCH = "BUILD_BRANCH"
```
Comment (wolfs) [inline build-logic/basics/src/main/kotlin/gradlebuild/basics/BuildParams.kt:59]: We probably shouldn't put those CI environment variables in the same place we put the feature switches for our build. Though I am not sure, yet.

What we should have is documentation for all the constants.

**Review by wolfs — APPROVED** [review]
I think this is already good as a first step. I'd like to find a way later to somehow have the constants and the type of parameters together.

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**